src/common.rs

```diff
@@ -202,11 +181,7 @@ impl<'gcc, 'tcx> ConstMethods<'tcx> for CodegenCx<'gcc, 'tcx> {
                 }
 
                 let value = self.const_uint_big(self.type_ix(bitsize), data);
-                if layout.value == Pointer {
-                    self.inttoptr(self.current_block.borrow().expect("block"), value, ty)
-                } else {
-                    self.const_bitcast(value, ty)
-                }
+                self.const_bitcast(value, ty)
```

Comment:
This does not look right. `bitsize` could potentially be `1`, so are you sure the type will be the same size?

Reply:
Only for a bool can bitsize be 1. A bool is represented as self.bool_type which is also 1 bit.

Reply:
Oh, right, `type_ix()` will return `self.i8_type` which is 8 bits, like `self.bool_type`.
In any case, it seems like the size is always equal, since the tests pass, but I'd also like an assert for the size.

Reply:
`get_size()` indeed only works on integers. If needed, I can make it work for pointer types.

That's a difference between LLVM and gcc: LLVM has a type `i1` of "1 bit" for bool while GCC's bool is 8 bits.

Reply:
`type_ix(1)` needs to be a `bool_type`. Seems like it is currently implemented as `i8_type`.

Reply:
Ok, can you add a TODO here saying to add an assert for the sizes, please?

Reply:
Done